ci/gitlab/.gitlab-ci.yml

```diff
@@ -0,0 +1,119 @@
+stages:
+  - Setup Docker
+  - Build and Test
+  
+variables:
+  # The docker images will be specific to the branch we are pushing to
+  UBUNTU_BIONIC_CI_IMAGE: ${CI_REGISTRY_IMAGE}:ci-${CI_COMMIT_REF_SLUG}-UBUNTU-BIONIC
+  DEBIAN_TESTING_CI_IMAGE: ${CI_REGISTRY_IMAGE}:ci-${CI_COMMIT_REF_SLUG}-DEBIAN_TESTING
+  GIT_SUBMODULE_STRATEGY: recursive
+  
+# Build the Docker image that will be used in subsequent CI/CD stages
+Setup Docker:
+  image: docker:latest
+  stage: Setup Docker
+  services:
+    - docker:dind
+  before_script:
+    - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
+  script:
+    - docker pull "$UBUNTU_BIONIC_CI_IMAGE" || true
+    - docker pull "${CI_REGISTRY_IMAGE}:ci-master" || true
+    - docker build --pull --cache-from "${CI_REGISTRY_IMAGE}:ci-master" --cache-from "$UBUNTU_BIONIC_CI_IMAGE" -f ${CI_PROJECT_DIR}/docker/Dockerfile.ubuntu.bionic -t "$UBUNTU_BIONIC_CI_IMAGE" .
+    - docker push "$UBUNTU_BIONIC_CI_IMAGE"
+    
+    - docker pull "$DEBIAN_TESTING_CI_IMAGE" || true
+    - docker pull "${CI_REGISTRY_IMAGE}:ci-master" || true
+    - docker build --pull --cache-from "${CI_REGISTRY_IMAGE}:ci-master" --cache-from "$DEBIAN_TESTING_CI_IMAGE" -f ${CI_PROJECT_DIR}/docker/Dockerfile.debian.testing -t "$DEBIAN_TESTING_CI_IMAGE" .
+    - docker push "$DEBIAN_TESTING_CI_IMAGE"
+    
+Test with p11 and coverage:
+  # Build and test aktualizr with p11 and generate coverage reports
+  variables:
+    TEST_BUILD_DIR: 'build-coverage'
+    TEST_CMAKE_BUILD_TYPE: 'Valgrind'
+    TEST_WITH_COVERAGE: '1'
+    TEST_WITH_P11: '1'
```

Comment:
We recently added `TEST_WITH_FAULT_INJECTION` here, which requires some external library (libfiu). Would be cool to add it.

Would be also helpful to have `TEST_SOTA_PACKED_CREDENTIALS` coming from a gitlab secret.